### PR TITLE
Milestone 2: Create and remove books

### PR DESCRIPTION
**Pull request by carloshs1994**
# Hi there 👋🏻 

## Sass

The project was built using Sass. The only difference was that I needed to add an additional loader to React, but the result is overall the same (A CSS output). It has the same dynamic response, it's just that the output looks more organized.

### Changes implemented in "create-and-remove-books" branch 

- ✅  My code follows the style guidelines of this project.
- ✅  Create Redux Store to manage state.
- ✅  You can add new books to the list.
- ✅  You can remove books from the list.

## Final notes

Is expected that this project meets the requirements. Please let me know if anything is left to be done or can be improved. Again, thanks for the review.

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
+  const bookAuthor = useRef(null);
+  const bookCategory = useRef(null);
+  const submitBooksToStore = (event) => {
+    event.preventDefault();
+    const newBook = {
+      title: bookTitle.current.value,
+      author: bookAuthor.current.value,
+      category: bookCategory.current.value,
+      id: bookTitle.current.value,
+    };
```
Comment (mikethreels) [inline src/components/Form.js]: - [x] in the requirements it is mentioned that the ID of the book is unique, right now if 2 books have the same title and the user deletes one of them both are deleted. below I've added the part of the assignment where this is mentioned:
```
// React component
...
// import useDispatch hook
import { useDispatch } from 'react-redux';
// import your Action Creators
import {addBook, removeBook} from './redux/books/books';

const dispatch = useDispatch();

...

const submitBookToStore = () => {
    const newBook = {
        id, // make sure it's unique
        title,
        author
    }

    // dispatch an action and pass it the newBook object (your action's payload)
    dispatch(addBook(newBook));
}

...

<button onClick={submitBookToStore}>Add Book</button>
```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Carlos, 

Great job, the code is very clean and I like the comments you include so you know what the code does 👍 !
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Form.js (hunk 1 above)
Comment (carloshs1994) [reply src/components/Form.js]: ## Hi @mikethreels 

Thanks for the feedback, I didn't consider that possibility. I fixed it now!

Comment (rloterh) [reply src/components/Form.js]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Carlos, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**